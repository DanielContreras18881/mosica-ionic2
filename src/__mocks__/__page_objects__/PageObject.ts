export class PageObject {
  constructor(public fixture) {}

  async updateAsync() {
    await flushPromises();
    this.fixture.detectChanges()
  }

  keepsSnapshot() {
    expect(this.text()).toMatchSnapshot()
  }

  text() {
    return this.removeComments(this.fixture.nativeElement.innerHTML.trim())
  }

  removeComments(text) {
    return text.replace(/<!--[\s\S]*?-->/g,'')
  }

  changeInput(inputName, inputValue) {
    this.fixture.componentInstance[inputName] = inputValue
    this.fixture.detectChanges()
  }

  containsText(text) {
    expect(this.text()).toContain(text)
  }
}

export function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}
