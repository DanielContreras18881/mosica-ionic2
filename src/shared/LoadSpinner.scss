.spinner {
  color: white;
  font-size: xx-large;
}
