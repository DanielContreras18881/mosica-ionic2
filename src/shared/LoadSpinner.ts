import {Component, Input} from '@angular/core';

@Component({
  selector: 'LoadSpinner',
  templateUrl: 'load-spinner.html',
})
export class LoadSpinner {
  @Input() isLoading
}
