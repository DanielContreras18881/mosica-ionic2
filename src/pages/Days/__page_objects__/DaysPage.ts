import {PageObject} from '../../../__mocks__/__page_objects__/PageObject';

export class DaysPage extends PageObject{
  constructor(public fixture) {
    super(fixture)
  }

  clickFirstGig() {
    const button = this.fixture.nativeElement.querySelector('.gig');
    button.click();
    this.fixture.detectChanges()
  }
}

export function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}
