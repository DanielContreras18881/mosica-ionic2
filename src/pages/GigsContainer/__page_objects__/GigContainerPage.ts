import {PageObject} from '../../../__mocks__/__page_objects__/PageObject';

export class GigsContainerPage extends PageObject{
  constructor(public fixture) {
    super(fixture)
  }

  async updateAsync() {
    await flushPromises();
    this.fixture.detectChanges()
  }

  keepsSnapshot() {
    expect(this.text()).toMatchSnapshot()
  }

  remove_line_breaks(string) {
    return string.replace(/(?:\r\n|\r|\n)/g, '<br />');
  }

  containsText(atext) {
    expect(this.text()).toContain(atext)
  }
}

export function flushPromises() {
  return new Promise(resolve => setImmediate(resolve))
}
