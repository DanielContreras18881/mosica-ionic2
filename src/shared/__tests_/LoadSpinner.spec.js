import {TestBed, async} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';
import { PageObject } from '../../__mocks__/__page_objects__/PageObject';
import { LoadSpinner } from '../LoadSpinner';

describe('LoadSpinner', () => {

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      schemas: [
        NO_ERRORS_SCHEMA
      ],
      declarations: [
        LoadSpinner
      ]
    }).compileComponents();
  }));


  it('snapshot when isLoading', async() => {
    let fixture = TestBed.createComponent(LoadSpinner);
    const page = new PageObject(fixture)
    page.changeInput('isLoading', true)
    page.keepsSnapshot()
  });

  it('snapshot when is NOT loading', async() => {
    let fixture = TestBed.createComponent(LoadSpinner);
    const page = new PageObject(fixture)
    page.changeInput('isLoading', false)
    page.keepsSnapshot()
  });
});

