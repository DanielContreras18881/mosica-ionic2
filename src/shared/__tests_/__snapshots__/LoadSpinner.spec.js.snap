// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadSpinner snapshot when is NOT loading 1`] = `""`;

exports[`LoadSpinner snapshot when isLoading 1`] = `
"<div class=\\"spinner\\">
  <p>Loading...</p>
</div>"
`;
